hive-metastore/src/main/java/org/apache/iceberg/hive/legacy/LegacyHiveTableUtils.java

```diff
@@ -56,7 +56,13 @@ static Schema getSchema(org.apache.hadoop.hive.metastore.api.Table table) {
     String schemaStr = props.get("avro.schema.literal");
     Schema schema;
     if (schemaStr != null) {
-      schema = AvroSchemaUtil.toIceberg(new org.apache.avro.Schema.Parser().parse(schemaStr));
+      org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(schemaStr);
+      org.apache.avro.Schema hiveSchema = LegacyHiveSchemaUtils.convertHiveSchemaToAvro(table);
```

Comment:
Good catch, I think I can change the signature of `convertHiveSchemaToAvro` to add this boolean flag parameter to `convertHiveSchemaToAvro(Table table, boolean mkFieldsOptional), so that this function directly return a non-null version of the schema.